Feat(board): add the redux implementation, for Electron

YvesLongchamp opened the pull request:
Here, you can found all of the boards
## feat(board): add the component, and container

## refactor: add actions and reducers

those are not working, for now. A fix is necessary

## fix(board): fix the actions and reducers

we still have to correct the config file

## refactor(board): change the actions, and reducers

change of the actions, to be usable for multiple boards
a change to do : add multiple boards for the board.tsx

## fix(board): add the store, and the route

## refactor: add the forgotten files

## feat: add the board component
(a simplist version)

## feat(board): add the tests
for the component, the reducers, and actions

## refactor: change some code, to be better written

YvesLongchamp committed:
feat(board): add the redux implementation, for Electron
Here, you can found all of the boards
feat(board): add the component, and container

refactor: add actions and reducers

those are not working, for now. A fix is necessary

fix(board): fix the actions and reducers

we still have to correct the config file

refactor(board): change the actions, and reducers

change of the actions, to be usable for multiple boards
a change to do : add multiple boards for the board.tsx

fix(board): add the store, and the route

refactor: add the forgotten files

feat: add the board component
(a simplist version)

feat(board): add the tests
for the component, the reducers, and actions

refactor: change some code, to be better written

Porquepix commented on package.json:
```diff
@@ -4,20 +4,38 @@
   "private": true,
   "dependencies": {
     "electron": "^1.7.9",
+    "enzyme": "^3.1.0",
```
All packages related to tests should be in dev part.

YvesLongchamp replied:
Will be fixed in the next commit.

Porquepix commented on src/components/Board/DnDContextBoard.tsx:
```diff
@@ -0,0 +1,14 @@
+
+import Board, { BoardProps } from './Board'
+
+const DnDContextBoard = (props2: BoardProps) => {
```
Why do we need a DnD ?

YvesLongchamp replied:
Drag and drop seems good, if you want to move the boards on a list. Keep the important ones before, etc...
Seems better to try to implement it now, and remove it if we do not need it.

YvesLongchamp committed:
fix: change the enzyme from dependencies to dev dependencies

YvesLongchamp committed:
fix: remove the DnD component

fix the board component, to be used by the container, and remove the DnD component for board

YvesLongchamp commented:
@Porquepix Changes are done, please confirm everything is good.
